src/python_testing/TC_OPSTATE_2_3.py

```diff
@@ -86,32 +85,34 @@ async def test_TC_OPSTATE_2_3(self):
         asserts.assert_equal(ret.commandResponseState.errorStateID, Clusters.OperationalState.Enums.ErrorStateEnum.kNoError,
                              "errorStateID(%s) should be NoError(0x00)" % ret.commandResponseState.errorStateID)
 
-        self.print_step(5, "Read OperationalState attribute")
-        operational_state = await self.read_mod_attribute_expect_success(endpoint=self.endpoint,
-                                                                         attribute=attributes.OperationalState)
-        logging.info("OperationalState: %s" % (operational_state))
-        asserts.assert_equal(operational_state, Clusters.OperationalState.Enums.OperationalStateEnum.kPaused,
-                             "OperationalState ID should be Paused(0x02)")
-
-        self.print_step(6, "Read CountdownTime attribute")
-        initial_countdown_time = await self.read_mod_attribute_expect_success(endpoint=self.endpoint,
-                                                                              attribute=attributes.CountdownTime)
-        logging.info("CountdownTime: %s" % (initial_countdown_time))
-        if initial_countdown_time is not NullValue:
-            in_range = (1 <= initial_countdown_time <= 259200)
-        asserts.assert_true(initial_countdown_time is NullValue or in_range,
-                            "invalid CountdownTime(%s). Must be in between 1 and 259200, or null " % initial_countdown_time)
-
-        self.print_step(7, "Waiting for 5 seconds")
-        time.sleep(5)
-
-        self.print_step(8, "Read CountdownTime attribute")
-        countdown_time = await self.read_mod_attribute_expect_success(endpoint=self.endpoint, attribute=attributes.CountdownTime)
-        logging.info("CountdownTime: %s" % (countdown_time))
-        asserts.assert_true(countdown_time != 0 or countdown_time == NullValue,
-                            "invalid CountdownTime(%s). Must be a non zero integer, or null" % countdown_time)
-        asserts.assert_equal(countdown_time, initial_countdown_time, "CountdownTime(%s) does not equal to the intial CountdownTime(%s)"
-                             % (countdown_time, initial_countdown_time))
+        if self.check_pics("OPSTATE.S.M.ST_PAUSED"):
```

Comment:
This does not make sense.  If the paused state is not supported, wouldn't step 4 ("Send Pause command") fail too?  The test plan certainly has a PICS condition on step 4 that is not reflected above.

However the test plan does _not_ have a PICS condition on step 5, apart from the attribute existing.  Is this PR implementing the test plan, or some proposed change to the test plan, or just something unrelated to the test plan?  If it's this last, the test plan needs to be updated either before or in parallel.

The Resume case has similar issues.

Reply:
@bzbarsky-apple  The test plan is also being updated: https://github.com/CHIP-Specifications/chip-test-plans/pull/3308/files

On lines 60 and 61 this script checks the pics which indicate whether the pause and resume commands are supported by the DUT, so step 4 is guarded.

OPSTATE.S.M.ST_PAUSED means "Does the DUT support testing the Paused(0x02) operational state?". The device may accept the Pause cmd and implement the (mandatory) OperationalState attribute but this may not necessarily mean that the OperationalState attribute will be set to Paused. So this is adding some finer granularity to the test's usage of pics codes. Please let us know if you think that requires a clarification comment in the test script or if it looks like overkill.

Reply:
This PICS code asks "Does the DUT support testing the Paused(0x02) operational state?". However, the spec explicitly specifies that if the `Pause` command is accepted, the `OperationlState` attribute should be set to the `Paused` state. So if step 4 can be tested, the device can only be in the `Paused` state, which is what this step tests for. Hence, I think that this `if` should also capture step 4.

Reply:
@plauric Thank you for the test plan PR link.  That link should have been in the initial PR description here to avoid wasting a bunch of reviewer time comparing the test changes to a test plan that is not what is being implemented.

That said, like @hicklin I'm not following how you can have PICS_SC_PAUSE (i.e. support the Pause command) but not have OPSTATE.S.M.ST_PAUSED.  If you support the Pause command, you have to support the paused state, no?

Reply:
The app running on the DUT may not have the logic fully implemented so it may accept the Pause command but not update the OperationalState attribute.

Reply:
> The app running on the DUT may not have the logic fully implemented so it may accept the Pause command but
> not update the OperationalState attribute.

But then it's not spec-compliant and should fail the test, no?

Reply:
Also: the line 60 check will fail the test outright if the Pause command is not supported.  That does not match the test plan either, as far as I can tell, even with the linked PR.

Reply:
> Also: the line 60 check will fail the test outright if the Pause command is not supported. That does not match the test plan either, as far as I can tell, even with the linked PR.

@bzbarsky-apple The test plan has a section called preconditions, under the description of this test (2.3). The preconditions are "OPSTATE.S.C00.Rsp(Pause) and OPSTATE.S.C03.Rsp(Resume)".

Reply:
@bzbarsky-apple After discussing with Duyen and William we concluded that if a Pause command is supported the Paused state must also be supported. Please see https://github.com/CHIP-Specifications/chip-test-plans/pull/3316